### PR TITLE
[docs][autoscaler] Add maintainers for node providers

### LINKED ISSUE #20974: [Feature][autoscaler] Include contacts for node provider maintainers in Ray docs.
### Search before asking

- [X] I had searched in the [issues](https://github.com/ray-project/ray/issues) and found no similar feature requirement.

### Description

The Ray documentation should include contacts from maintainers of node providers.
The contacts for Aliyun NodeProvider maintainers are already there and we could follow the same pattern for the rest.
https://docs.ray.io/en/latest/cluster/cloud.html#ray-with-cloud-providers

### Use case

_No response_

### Related issues

_No response_

### Are you willing to submit a PR?

- [X] Yes I am willing to submit a PR!

### PR DESCRIPTION
**Pull request by wuisawesome**
## Why are these changes needed?
This PR adds documentation for the maintainers of the various node providers. 

## Related issue number
Closes #20974

## Checks

- [ ] I've run `scripts/format.sh` to lint the changes in this PR.
- [ ] I've included any doc changes needed for https://docs.ray.io/en/master/.
- [ ] I've made sure the tests are passing. Note that there might be a few flaky tests, see the recent failures at https://flakey-tests.ray.io/
- Testing Strategy
   - [ ] Unit tests
   - [ ] Release tests
   - [ ] This PR is not tested :(

### CODE REVIEW

**Comment by wuisawesome** [comment]
cc @zhe-thoughts @AmeerHajAli @richardliaw

File: doc/source/cluster/cloud.rst [hunk 1]
```diff
@@ -46,6 +46,8 @@ Ray with cloud providers
         $ ray down ray/python/ray/autoscaler/aws/example-full.yaml
 
 
+    AWS Node Provider Maintainer: pdames, Zyiqin-Miranda, DmitriGekhtman, wuisawesome
```
Comment (DmitriGekhtman) [inline doc/source/cluster/cloud.rst:49]: Should indicate that these are GitHub handles, like
"AWS Node Provider Maintainer GitHub handles"
or
"AWS Node Provider Maintainers (GitHub handles)"
or sth

Comment (wuisawesome) [reply doc/source/cluster/cloud.rst]: good idea

**Comment by wuisawesome** [comment]
> I haven't looked at how other OSS projects do this. But one question is how to keep the info updated

Good question, one data point is open telemetry where collectors are put in a separate repo and maintainers are listed in the git repo. 

https://github.com/open-telemetry/opentelemetry-collector

**Review by DmitriGekhtman — APPROVED** [review]
Github -> GitHub

**Pull request merged**